ls1intum/Artemis
# `Development`: Update client dependencies to Angular 15 and Jest 29

### Checklist
#### General



### Motivation and Context
Artemis currently has Angular v.14. We want to update Angular client to the latest major v.15, as the major releases can include the following benefits for the evolution of the angular framework:
* Release of useful new features.
* Better performance.

Moreover, each major version support usually lasts 18 months, with 6 months active support phase and 12 months LTS. During this time, the critical issues are fixed, and security patches are released.

Artemis also uses Jest Testing Framework v.28, which we want to update to v. 29. 

### Description
* Using the official [upgrade guide](https://update.angular.io/?l=3&v=14.0-15.0) we upgrade the Angular v. 14 to v. 15.
* Using the official [upgrade guide](https://jestjs.io/docs/upgrading-to-jest29) we upgraded the Jest v. 28 to v. 29

### Steps for Testing

Prerequisites:
- 1 admin


Check the angular client, if everything works as expected after the update:
* Check all components as a user 
* Check all tests run: 
* Check if build is succeed using the following command in terminal:

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test
